paramak/parametric_shapes/sweep_circle_shape.py

```diff
@@ -0,0 +1,125 @@
+from collections import Iterable
+
+import cadquery as cq
+
+from paramak import Shape
+
+
+class SweepCircleShape(Shape):
+    """Sweeps a 2D circle of a defined radius along a defined spline path to create a
+    3D CadQuery solid.
+
+    Args:
+        radius (float): Radius of 2D circle to be swept.
+        path_points (list of tuples each containing X (float), Z (float)): A list of XY,
+            YZ or XZ coordinates connected by spline connections which define the path
+            along which the 2D shape is swept.
+        path_workplane (str): Workplane in which the spline path is defined.
```

Comment:
Some defaults value appear to be missing. Similarly, since arguments workplane, stp_filename, stl_filename aren't set to defaults here, I think it should appear in the docstrings. Don't you think ? That's what I did in other shapes so if you guys want want to keep this style I should make a few changes in the other components